Makefile

```diff
@@ -24,7 +24,7 @@ ifdef GH_OAUTH_CLIENT_SECRET
 endif
 
 bin/gh: $(BUILD_FILES)
-	go build -trimpath -ldflags "${GO_LDFLAGS}" -o "$@" ./cmd/gh
+	CGO_ENABLED=0 go build -trimpath -ldflags "${GO_LDFLAGS}" -o "$@" ./cmd/gh
```

Comment:
The Makefile is only used in development, where I feel that CGO should be allowed.

Would it make more sense to move this to our goreleaser config file, which is used for creating our precompiled binaries?

I'm thinking it should also only apply to Linux binaries, as we plan to compile binaries for macOS and Windows natively from their respective platforms while having CGO enabled.

Reply:
I thought it was rather unusual to use CGO (it's a strange default). At CircleCI (where I work) we disable CGO for all platforms (Linux, Mac, Windows) on our GoLang based build-agent, for example.

Reply:
Of course you're right that it should also go in the goreleaser config!

I try and keep the release config as similar to dev config as possible, myself, so I'd have this setting in both places, myself.

Reply:
At any rate, I've updated a few more places now - let me know what you think @mislav !

Reply:
Another example: see `goreleaser`'s own `.goreleaser` file: https://github.com/goreleaser/goreleaser/blob/master/.goreleaser.yml

Reply:
> unless you specifically required an operating-system provided library of some kind

We do plan to support macOS Keychain and the equivalent encrypted credentials store on Windows, and for that we will need to enable CGO and build on the same platform. Furthermore, it looks like some network issues (such as automatically picking up the system proxy settings) are also resolved with those builds. Lastly, we will need to code-sign all of our Windows and macOS executables, so cross-compilation for those platforms from Linux will not be as viable anymore.

For all platforms other than macOS/Windows, I agree that CGO should be disabled. 👍

Reply:
In case it wasn't clear, I still maintain my position that `CGO_ENABLED=0` does not belong in the Makefile. In the future, some developer may try to link to some OS-provided library in an experimental branch, and `make bin/gh` will not work for them. Since Makefile is only used for development, let's have the defaults be the defaults. Developers may set
 `CGO_ENABLED=0` themselves in their environment.

Reply:
For sure I agree with that if you need to access platform-specific libraries! Otherwise dev builds won't work.

I'll change this to do the switch only on Linux.

Reply:
I've changed the CI to only do CGO=0 on the minimal ubuntu build (so we have a check if this will compile on Linux in PRs), and to apply the GCO-0 only on Linux releases

Reply:
Is there anything else you'd like me to change @mislav?